src/symbol/frame.rs

```diff
@@ -0,0 +1,264 @@
+use std::fmt;
+
+use super::dwarf_utils::SearchAction;
+use super::*;
+
+pub struct Frame<'a> {
+    dwarf: &'a gimli::Dwarf<Reader<'a>>,
+    unit: Option<&'a gimli::Unit<Reader<'a>>>,
+    frame: addr2line::Frame<'a, Reader<'a>>,
+}
+
+impl<'a> Frame<'a> {
+    pub fn function_debuginfo(
+        &self,
+    ) -> Option<(
+        &'a gimli::Dwarf<Reader<'a>>,
+        &'a gimli::Unit<Reader<'a>>,
+        gimli::UnitOffset<usize>,
+    )> {
+        self.frame
+            .dw_die_offset
+            .map(|unit_offset| (self.dwarf, self.unit.unwrap(), unit_offset))
+    }
+
+    pub fn print_debuginfo(&self) {
+        println!(
```

Comment:
Later, we probably can replace this with `impl<'a> fmt::Display for Frame<'a>`, then it can be printed into a string.